teuthology/task/install.py

```diff
@@ -963,8 +963,22 @@ def upgrade_common(ctx, config, deploy_style):
             # FIXME: again, make extra_pkgs distro-agnostic
         pkgs += extra_pkgs
 
+        installed_version = packaging.get_package_version(remote, 'ceph-common')
+        upgrade_version = _get_gitbuilder_project(ctx, remote, node).version
+        log.info("Ceph {s} upgrade from {i} to {u}".format(
+            s=system_type,
+            i=installed_version,
+            u=upgrade_version
+        ))
+        if installed_version > upgrade_version:
```

Comment:
Ah, but I see your point - '9.0.0' will indeed be greater than '10.0.0'

OK, back to the drawing board :-)